eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1517,21 +1518,19 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, remo
       log.info(s"txid=${tx.txid} has reached mindepth, updating closing state")
       context.system.eventStream.publish(TransactionConfirmed(d.channelId, remoteNodeId, tx))
       // first we check if this tx belongs to one of the current local/remote commits, update it and update the channel data
-      val d1 = d.copy(
-        localCommitPublished = d.localCommitPublished.map(localCommitPublished => {
-          // If the tx is one of our HTLC txs, we now publish a 3rd-stage claim-htlc-tx that claims its output.
-          val (localCommitPublished1, claimHtlcTx_opt) = Closing.claimLocalCommitHtlcTxOutput(localCommitPublished, keyManager, d.commitments, tx, nodeParams.onChainFeeConf.feeEstimator, nodeParams.onChainFeeConf.feeTargets)
-          claimHtlcTx_opt.foreach(claimHtlcTx => {
-            txPublisher ! PublishFinalTx(claimHtlcTx, claimHtlcTx.fee, None)
-            blockchain ! WatchTxConfirmed(self, claimHtlcTx.tx.txid, nodeParams.channelConf.minDepthBlocks)
-          })
-          Closing.updateLocalCommitPublished(localCommitPublished1, tx)
-        }),
-        remoteCommitPublished = d.remoteCommitPublished.map(Closing.updateRemoteCommitPublished(_, tx)),
-        nextRemoteCommitPublished = d.nextRemoteCommitPublished.map(Closing.updateRemoteCommitPublished(_, tx)),
-        futureRemoteCommitPublished = d.futureRemoteCommitPublished.map(Closing.updateRemoteCommitPublished(_, tx)),
-        revokedCommitPublished = d.revokedCommitPublished.map(Closing.updateRevokedCommitPublished(_, tx))
-      )
+      val d1 = d
+        .modify(_.localCommitPublished.each).using { localCommitPublished =>
```

Comment:
I'm not a big fan of this, it's really not obvious what `.each` means, I actually find the code harder to understand now...